src/syzygy/tbprobe.cpp

```diff
@@ -1708,26 +1732,27 @@ void* init(Entry& e, const Position& pos) {
     return e.baseAddress;
 }
 
+template<typename T>
+T result_to_score(Value value) {
+    if (value > 0)
+        return std::is_same<T, WDLScore>::value ? T(WDLWin) : T(1);
+    else if (value < 0)
+        return std::is_same<T, WDLScore>::value ? T(WDLLoss) : T(-1);
+    else
+        return T(WDLDraw);
+}
+
 template<typename E, typename T = typename Ret<E>::type>
 T probe_table(const Position& pos, ProbeState* result, WDLScore wdl = WDLDraw) {
 
-#ifdef ATOMIC
-    if (pos.is_atomic()) {
-        if (pos.is_atomic_loss())
-            return std::is_same<T, WDLScore>::value ? T(WDLLoss) : T(-1);
-        if (pos.is_atomic_win())
-            return std::is_same<T, WDLScore>::value ? T(WDLWin) : T(1);
-    }
-#endif
+    // Check for variant end
+    if (pos.is_variant_end())
+        return result_to_score<T>(pos.variant_result());
+
+    // Check for checkmate and stalemate
+    if (MoveList<LEGAL>(pos).size() == 0)
+        return result_to_score<T>(pos.checkers() ? pos.checkmate_value() : pos.stalemate_value());
 
-#ifdef ANTI
-    if (pos.is_anti()) {
-        if (pos.is_anti_loss())
-            return std::is_same<T, WDLScore>::value ? T(WDLLoss) : T(-1);
-        if (pos.is_anti_win())
-            return std::is_same<T, WDLScore>::value ? T(WDLWin) : T(1);
-    } else
```

Comment:
Thanks, I missed that. I think this is difficult to generalize, since the condition is valid for standard chess, atomic, and three-check, but not for other variants. My idea would be to write a method to check for insufficent material (at least for the one with minimal material), i.e. K vs. K in standard and atomic chess, and opposite colored bishops in giveaway/suicide (is this essential?). What do you think?